logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/migration/MigrateFromPersonalToTeamUseCase.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.user.migration
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.fold
+
+/**
+ * Use case to migrate user personal account to team account.
+ * This needs at least API V7 to work.
+ */
+interface MigrateFromPersonalToTeamUseCase {
+    suspend operator fun invoke(teamName: String): MigrateFromPersonalToTeamResult
+}
+
+sealed class MigrateFromPersonalToTeamResult {
+    data class Success(val teamId: String, val teamName: String) : MigrateFromPersonalToTeamResult()
```

Comment:
Do you need to pass to the app `teamId` and `teamName` ?, the app is already having those

Reply:
Removed id as we don't use it. Left the team name and updated the state just in case API somehow changed the name

Reply:
Yeah but API should not change a name provided by the user